### ISSUE TITLE
Need to improve performance compared to TCP

### ISSUE THREAD

**Issue by glen-dai** [lucas-clemente/quic-go#790]
I'v done some benchmark with quic-go, TCP, goquic(https://github.com/devsisters/goquic) to donwnload a 600MB byte file under RTT of 192ms, no packet loss。TCP and goquic reaches 14MB/s, while quic-go only gets 7.5MB/s。goquic has cwd ajust to about 10M byte。 I also try to modify quic-go's parameters( ReceiveStreamFlowControlWindow from 32KB to 4MB, ReceiveConnectionFlowControlWindow from 48KB to 10MB, and InitialCongestionWindow from 32 to 200), but makes no difference.

Am I do not do it correctly or quic-go really need to improve the performance?

**Comment by lucas-clemente** [comment]
Yes, we need to put more work into improving our performance. I don't have a timeline, sorry.

**Comment by sean-** [comment]
We performed a similar test using the `quicc` and `quicd` proxies found here and were somewhat discouraged by the results.

https://github.com/sean-/2017-joyent-hackathon

We also found that downloading a 100MB file failed w/ `quic-go` but succeeded with a 10MB file.  We ran out of time to investigate further.

heh, naturally, pprof picked up flame graph support earlier today and would have been useful during this exercise as it could have easily provided some guidance as to where the problems were, if they're CPU or in the CC (we were surprised to find that this QUIC impl didn't implement BBR):

https://github.com/google/pprof
https://github.com/google/pprof/pull/188
